app/models/foreman_fog_proxmox/proxmox_vm_commands.rb

```diff
@@ -55,7 +55,9 @@ def create_vm(args = {})
 
     def destroy_vm(uuid)
       vm = find_vm_by_uuid(uuid)
-      unless vm.nil?
+      if vm.nil?
```

Comment:
More ruby like:
```
return true if vm.nil?

vm.stop if vm.ready?
vm.destroy
```

maybe think also about using a try/catch block. Verify how this is done for other compute resources, too.

Reply:
For other compute resources if they cannot find the vm then they throw `ActiveRecord::RecordNotFound` error which is caught by rescue condition. But in proxmox if the vm is not found it returns `nil` which cannot be caught by try/catch method.

Reply:
thanks.